### ISSUE TITLE
[MPH-154] The output of the plugin should be flushed when using forceStdout

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-help-plugin#289]
**[Patrick van Rhijn](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pietje)** opened **[MPH-154](https://issues.apache.org/jira/browse/MPH-154?redirect=false)** and commented

When you use the plugin the output doesn't contain a line-ending making the output hard to use in scripting when you have JAVA_OPTIONS defined.

In the current situation the plugin gives me the following output: 

$ mvn -N -q org.apache.maven.plugins:maven-help-plugin:3.1.0:evaluate -Dexpression='project.groupId' -DforceStdout=true
**nl.foo.barPicked up _JAVA_OPTIONS: -XX:+ForceTimeHighResolution -Dsun.java2d.noddraw=true -Duser.timezone='Europe/Amsterdam'**

It would be better if the output would be divided over two lines and look like the output of the exec-maven-plugin:

$ mvn -N -q -Dexec.executable='echo' -Dexec.args='${project.groupId}' org.codehaus.mojo:exec-maven-plugin:1.4.0:exec
**nl.foo.bar**
**Picked up _JAVA_OPTIONS: -XX:+ForceTimeHighResolution -Dsun.java2d.noddraw=true -Duser.timezone='Europe/Amsterdam'**

 

---

**Issue Links:**
- [MPH-155](https://issues.apache.org/jira/browse/MPH-155) quiet output to stdout doesn't always work

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

The intention was exactly to have the output does **not** contain a line ending cause it makes usage in scripts easier like:

```
RESULT=$(mvn -N -q org.apache.maven.plugins:maven-help-plugin:3.1.0:evaluate -Dexpression='project.groupId' -DforceStdout=true)
```

If it would contain a line ending this had meant you have to strip the line feed in situations like the above..
In the current implementation you can simply use it:

```
echo $RESULT
```

Apart from that Can you explain what you are trying to achieve ? Do you have an example pom which shows the above results? The output `Picked up _JAVA_OPTIONS` is only known to me inside Jenkins which is done by the pipeline spy integration....

**Comment by jira-importer** [comment]
**[Patrick van Rhijn](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pietje)** commented

I'm looking into the fact that the line "Picked up..." is generated by the jvm and is specific to the jvm. The result differs when you use a zulu, openjdk or a oracle jre. So I'm looking into a way to suppress this message on the OS.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

As I already asked where you have got that output from ?

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

I've been looking at this with [Patrick van Rhijn](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pietje) and we have to adjust the analysis. The problem is that with a PrintStream every EOL implies a flush, but a close doesn't have to flush. In this case it could be that nothing will be written to the output stream.

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

Fixed in [26224aa40d6a8dcd383ecdb823d1a60d1bb9b485](https://gitbox.apache.org/repos/asf?p=maven-help-plugin.git;a=commit;h=26224aa40d6a8dcd383ecdb823d1a60d1bb9b485)
